functions/Copy-DbaCredential.ps1

```diff
@@ -117,37 +118,54 @@ function Copy-DbaCredential {
+            return
+        }
+        Invoke-SmoCheck -SqlInstance $sourceServer
+
+        if ($null -ne $SourceSqlCredential.Username) {
+            Write-Message -Level Verbose -Message "You are using SQL credentials and this script requires Windows admin access to the $Source server. Trying anyway."
+        }
 
-                .OUTPUT
-                    System.Data.DataTable
-            #>
-            [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidUsingPlainTextForPassword", "", Justification = "For Credentials")]
-            param (
-                [bool]$Force
-            )
+        Write-Message -Level Verbose -Message "Decrypting all Credential logins and passwords on $($sourceServer.Name)"
+        try {
+            $decryptedCredentials = Get-DecryptedObject -SqlInstance $sourceServer -Type Credential -EnableException
+        } catch {
+            Stop-Function -Message "Failed to decrypt credentials on $($sourceServer.Name)" -ErrorRecord $_
+            return
+        }
 
-            Write-Message -Level Verbose -Message "Collecting Credential logins and passwords on $($sourceServer.Name)"
+        Write-Message -Level Verbose -Message "Getting all Credentials that should be processed on $($sourceServer.Name)"
+        $credentialList = Get-DbaCredential -SqlInstance $sourceServer -Name $Name -ExcludeName $ExcludeName -Identity $Identity -ExcludeIdentity $ExcludeIdentity
+    }
+    process {
+        if (Test-FunctionInterrupt) { return }
+
+        foreach ($destinstance in $Destination) {
             try {
-                $sourceCredentials = Get-DecryptedObject -SqlInstance $sourceServer -Type Credential -EnableException
+                $destServer = Connect-DbaInstance -SqlInstance $destinstance -SqlCredential $DestinationSqlCredential -MinimumVersion 9
             } catch {
-                Stop-Function -Message "Failed to decrypt credentials on $($sourceServer.Name)" -ErrorRecord $_
-                return
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $destinstance -Continue
             }
-
-            $credentialList = Get-DbaCredential -SqlInstance $sourceServer -Name $Name -ExcludeName $ExcludeName -Identity $Identity -ExcludeIdentity $ExcludeIdentity
+            Invoke-SmoCheck -SqlInstance $destServer
 
             Write-Message -Level Verbose -Message "Starting migration"
-            foreach ($credential in $credentialList) {
-                $destServer.Credentials.Refresh()
-                $credentialName = $credential.Name
+            $destServer.Credentials.Refresh()
+            foreach ($cred in $credentialList) {
+                $credentialName = $cred.Name
 
                 $copyCredentialStatus = [pscustomobject]@{
-                    SourceServer      = $sourceServer.Name
-                    DestinationServer = $destServer.Name
+                    SourceServer      = $sourceServer.DomainInstanceName
```

Comment:
Should this be ComputerName instead? I feel like there are sometimes problems with DomainInstanceName. Perhaps it's not always filled out or accurate.

Reply:
`SourceServer` is used for the output object in all the Copy commands @potatoqualitee

Reply:
Oh, you are referring to property name? If that, yes we use ComputerName for the value in other commands.

Reply:
Thank you for double checking 😊

Reply:
Also looks like `Connect` object does checks for populating `ComputerName`

https://github.com/dataplat/dbatools/blob/3d8d330f001853268c78aa0eebb2aef10ae9630c/functions/Connect-DbaInstance.ps1#L1573-L1592

Reply:
No, we don't need the computername here, but "SqlInstance". This is the output from the currently released code:

![image](https://user-images.githubusercontent.com/66946165/152292995-c732b752-7bcc-42e9-b7ab-a2da68e10c18.png)

And we mostly fill "SqlInstance" using ".DomainInstanceName", just search for "SqlInstance  = $server.DomainInstanceName" in the code.

We can not use ".Name" of "$server", as this has the prefix "ADMIN:" when using a DAC - and I think we don't want to have that in the output.

Reply:
ahhhh make sense, thanks